From d428c8c12fb3703200986667672c48c28d788306 Mon Sep 17 00:00:00 2001
From: Elliot Chen <elliot.chen@nxp.com>
Date: Thu, 4 May 2023 17:00:51 +0800
Subject: [PATCH 09/13] MMFMWK-9239 multiqueue: support query buffering range
 if use interleave

Upstream-Status: Pending

Signed-off-by: Elliot Chen <elliot.chen@nxp.com>
Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 plugins/elements/gstmultiqueue.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/plugins/elements/gstmultiqueue.c b/plugins/elements/gstmultiqueue.c
index 34ca31e..196ae53 100644
--- a/plugins/elements/gstmultiqueue.c
+++ b/plugins/elements/gstmultiqueue.c
@@ -2857,6 +2857,14 @@ gst_multi_queue_sink_query (GstPad * pad, GstObject * parent, GstQuery * query)
         }
       }
       gst_query_set_buffering_percent (query, mq->use_interleave, percent);
+
+      /* support query buffering range if use interleave */
+      GstFormat format;
+      gst_query_parse_buffering_range (query, &format, NULL, NULL, NULL);
+      if ((format == GST_FORMAT_TIME) && mq->use_interleave) {
+        gst_query_set_buffering_range (query, format, sq->src_segment.position,
+          sq->sink_segment.position, mq->min_interleave_time);
+      }
       res = TRUE;
       break;
     }
-- 
2.49.0

